rmf_site_editor/src/interaction/cursor.rs

```diff
@@ -129,13 +129,12 @@ impl Cursor {
     ) {
         self.remove_preview(commands);
         self.preview_model = if let Some(model_instance) = model_instance {
-            if let Some(mut spawn_instance) =
-                model_loader.spawn_model_instance(self.frame, model_instance.clone(), None)
-            {
-                Some(spawn_instance.insert(Pending).id())
-            } else {
-                None
-            }
+            Some(
```

Comment:
Not a big deal but I think this can all be rewritten as

```rust
self.preview_model = model_instance
    .and_then(|model_instance| 
        model_loader.spawn_model_instance(self.frame, model_instance)
        .insert(Pending)
        .id()
    );
```

Reply:
[6067fef](https://github.com/open-rmf/rmf_site/pull/242/commits/6067feffabf66bd01dcb2e8c3537e8937bb041fb)